server/plugin/api.go

```diff
@@ -195,6 +199,77 @@ func (p *MatterpollPlugin) handleSubmitDialogRequest(handler submitDialogHandler
 	}
 }
 
+func (p *MatterpollPlugin) handleCreatePoll(vars map[string]string, request *model.SubmitDialogRequest) (*i18n.Message, *model.SubmitDialogResponse, error) {
+	publicLocalizer := p.getServerLocalizer()
+	creatorID := request.UserId
+
+	question, ok := request.Submission[questionKey].(string)
+	if !ok {
+		return commandErrorGeneric, nil, errors.Errorf("failed to get submission key: %s", questionKey)
+	}
+
+	var answerOptions []string
+	o1, ok := request.Submission["option1"].(string)
+	if !ok {
+		return commandErrorGeneric, nil, errors.Errorf("failed to get submission key: %s", questionKey)
+	}
+	answerOptions = append(answerOptions, o1)
+
+	o2, ok := request.Submission["option2"].(string)
+	if !ok {
+		return commandErrorGeneric, nil, errors.Errorf("failed to get submission key: %s", questionKey)
+	}
+	answerOptions = append(answerOptions, o2)
+
+	o3, ok := request.Submission["option3"].(string)
+	if ok {
+		answerOptions = append(answerOptions, o3)
+	}
+
+	var settings []string
+	for k, v := range request.Submission {
+		if strings.HasPrefix(k, "setting-") {
+			b, ok := v.(bool)
+			if b && ok {
+				settings = append(settings, strings.TrimPrefix(k, "setting-"))
+			}
+		}
+	}
+
+	userLocalizer := p.getUserLocalizer(creatorID)
+	poll, errMsg := poll.NewPoll(creatorID, question, answerOptions, settings)
+	if errMsg != nil {
+		response := &model.SubmitDialogResponse{
+			Error: p.LocalizeErrorMessage(userLocalizer, errMsg),
+		}
+		return nil, response, nil
+	}
+
+	if err := p.Store.Poll().Save(poll); err != nil {
+		return commandErrorGeneric, nil, errors.Wrap(err, "failed to save poll")
+	}
+
+	displayName, appErr := p.ConvertCreatorIDToDisplayName(creatorID)
+	if appErr != nil {
+		return commandErrorGeneric, nil, errors.Wrap(appErr, "failed to get display name for creator")
+	}
+
+	actions := poll.ToPostActions(publicLocalizer, manifest.ID, displayName)
+	post := &model.Post{
+		UserId:    p.botUserID,
+		ChannelId: request.ChannelId,
+		RootId:    request.CallbackId,
+		Type:      model.POST_DEFAULT,
```

Comment:
```suggestion
		Type:      MatterpollPostType,
		Props: map[string]interface{}{
			"poll_id": poll.ID,
		},
```

Please use custom post type for ExperimentalUI.

Reply:
Good catch :+1: